### ISSUE TITLE
Support user provided annotations

### ISSUE THREAD

**Issue by TamalSaha** [appscode/voyager#103]
As #100 notes, it will be useful for users to be able to apply annotations on HAproxy service and/or Deployment. Users can give a configmap with their custom annotations that we can use. Users can apply the following annotation to ingress: 
`ingres.appscode.com/annotations: <user-provided-cfgmap-name>`

```
data:
  svc: |
  {
    "a/s1": "v1",
    "b/s2": "v2",
  }
  pods: |
  {
    "u/up1": "v1",
    "w/p2": "v2",
  }
kind: ConfigMap
metadata:
  name: <user-provided-cfgmap-name>
  namespace: <same-as-ingress>
```
The `json` object under `svc` key will be applied to service running in front of HAproxy. The `json` object under `pods` key will be applied to pods (Deployment/ DaemonSet) running HAproxy. This will allow users to use full spectrum of keys in annotations.

The keys will be filtered to remove any `ingress.appscode.com/` keys, since those should be applied to `ingress` directly.

We should watch these configmaps so that they are dynamically updated. This can be done using a `UpdateAnnotations` mode.

I think the configmaps should be the vector for user annotations. Using secrets does not make sense, since annotations will be ultimately visible to users.

**Comment by TamalSaha** [comment]
I also wonder, if we should create a new TPR type to avoid the restrictions on `key` in configmaps. For example, configmaps do not allow `/` in their keys. That is why I proposed the `json` style format. But if we use our own TPR, we can only restrict to charset allowed in annotation keys.

**Comment by sadlil** [comment]
This feature was implemented in #105 via ingress annotation.

`ingress.appscode.com/annotations.service` and `ingress.appscode.com/annotations.pod` can be added in Ingress annotations. The values for those are json encoded annotation data, that will be applied into service and pods' annotations respectively.
